testdata/project-v2/_output/config-gen.yaml

```diff
@@ -0,0 +1,392 @@
+apiVersion: v1
```

Comment:
Also, to imo we need to ensure that the projects using config-gen will work well as expected on the cluster. Then, we need to run the same tests https://github.com/kubernetes-sigs/kubebuilder/blob/master/test/e2e/v3/plugin_cluster_test.go. 

IMO we need to copy and paste the go/v3 scenario and tests and generate it with config-gen. Then, we comment all tests over features/configs which is not covered yet and add todo(s) for that. In this way, we will know what still missing and we will able to ensure that it is working.

Reply:
I like the idea of having e2e tests.  We have unit tests right now.  Lets merge this and add the e2e's before moving it out of alpha.